sematic/scheduling/job_details.py

```diff
@@ -0,0 +1,234 @@
+# Standard Library
+import time
+from dataclasses import dataclass, field
+from typing import List, Literal, Optional
+
+# TODO: remove ExternalJob and KubernetesExternalJob once they are replaced by this & Job
+
+
+# This is not an Enum because dataclasses.asdict doesn't produce
+# a json-serializable result for enums.
+class KubernetesJobState:
+    """Simple strings describing the K8s job state.
+
+    Though it is meant to be associated with the *job* it draws
+    from details relating to the *pod*, as the latter is more
+    information-rich. A "Job" may show up as active for both a pending
+    and a running pod, for example.
+
+    Drawn from the docs for pod lifecycle:
+    https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
+
+    Some additional states are added to represent the extra tracking
+    Sematic is performing. Extra states not covered in pod lifecycle
+    include:
+    - Requested
+    - Deleted
+    - Restarting
+    - Terminated
+
+    Attributes
+    ----------
+    Requested:
+        Sematic has requested the job, but Kubernetes has not yet assigned a status
+        to it.
+    Pending:
+        The pod for the job is in the "Pending" phase.
+    Running:
+        The pod for the job is in the "Running" phase.
+    Restarting:
+        The job is in an intermediate state where one pod is being removed
+        while another is being created.
+    Succeeded:
+        The job has at least one pod which has exited with a 0 status.
+    Failed:
+        The job ended with no pods in a succeeded state.
+    Deleted:
+        The job once existed, but does no longer.
+    """
+
+    def __init__(self) -> None:
+        raise RuntimeError(
+            "This is meant to emulate an Enum and not to be instantiated directly"
+        )
+
+    Requested = "Requested"
+    Pending = "Pending"
+    Running = "Running"
+    Restarting = "Restarting"
+    Succeeded = "Succeeded"
+    Failed = "Failed"
+    Deleted = "Deleted"
+
+    @classmethod
+    def is_active(cls, state) -> bool:
+        return state in _ACTIVE_STATES
+
+
+_ACTIVE_STATES = {
+    KubernetesJobState.Requested,
+    KubernetesJobState.Pending,
+    KubernetesJobState.Running,
+    KubernetesJobState.Restarting,
+}
+
+KubernetesJobStateString = Literal[
+    "Requested",
+    "Pending",
+    "Running",
+    "Restarting",
+    "Succeeded",
+    "Failed",
+    "Deleted",
+]
+
+
+class JobKind:
+    def __init__(self) -> None:
+        raise RuntimeError("This is meant to emulate an enum, not be instantiated.")
+
+    resolver = "resolver"
+    run = "run"
+
+
+JobKindString = Literal[
+    "resolver",
+    "run",
+]
+
+
+@dataclass
+class PodSummary:
```

Comment:
All these classes are Kubernetes-specific, so please make sure each class contains `Kubernetes` in its name, and that the file contains `kubernetes` in its name.